python-sdk/src/astro/files/locations/sftp/sftp.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import annotations
+
+import os
+from urllib.parse import urlparse, urlunparse
+
+from airflow.providers.sftp.hooks.sftp import SFTPHook
+
+from astro.constants import FileLocation
+from astro.files.locations.base import BaseFileLocation
+
+
+class SFTPLocation(BaseFileLocation):
+    """Handler SFTP object store operations"""
+
+    location_type = FileLocation.SFTP
+
+    @property
+    def hook(self) -> SFTPHook:
+        return SFTPHook(ssh_conn_id=self.conn_id) if self.conn_id else SFTPHook()
+
+    @property
+    def transport_params(self) -> dict:
+        """get SFTP credentials for remote file system"""
+        client = self.hook.get_connection(self.conn_id)
+        extra_options = client.extra_dejson
+        if "key_file" in extra_options:
+            key_file = extra_options.get("key_file")
+            return {"connect_kwargs": {"key_filename": key_file}}
+        elif client.password:
+            return {"connect_kwargs": {"password": client.password}}
+        return {}
+
+    @property
+    def paths(self) -> list[str]:
+        """Resolve SFTP file paths with prefix"""
+        url = urlparse(self.path)
```

Comment:
We're parsing this multiple times in each of the properties. Can we have a parsed url as an instance variable and reuse it?